### PR TITLE
Add GitHub CODEOWNERS file

### PR DESCRIPTION
**Pull request by dhruvmanila**
## Summary

Add GitHub CODEOWNERS file.

Initiating this, we can discuss and iterate further.

https://help.github.com/articles/about-codeowners/

## Test Plan

Look out for review requests :)

### CODE REVIEW

File: .github/CODEOWNERS
Comment (charliermarsh) [inline .github/CODEOWNERS:87]: What's the process by which this is kept up-to-date? (E.g., if we add new directories, or rename, or remove?)

Comment (dhruvmanila) [reply .github/CODEOWNERS:87]: I'm seeing a message on top of the file saying "This CODEOWNERS file is valid". I believe GitHub checks it against the repository. Let me do a quick commit and verify.

If not, we could use https://github.com/mszostok/codeowners-validator although it seems to be not maintained anymore.

Comment (dhruvmanila) [reply .github/CODEOWNERS:87]: Hmm, I think GitHub seems to be only checking for syntax. Also, the `codeowners-validator` is giving me segmentation fault :(

Comment (dhruvmanila) [reply .github/CODEOWNERS:87]: The path could contain patterns as well but if we don't use patterns then we could do a manual check on just the existence of path in CI.

Comment (charliermarsh) [reply .github/CODEOWNERS:87]: In that case, I might vote to only include the folders for which we're enforcing rules, and add them as we go. What do you think?

Comment (dhruvmanila) [reply .github/CODEOWNERS:87]: Yeah, makes sense. Do you have any other folders in mind? (I'll comment out the rest before merging)

Comment (charliermarsh) [reply .github/CODEOWNERS:87]: Na, that's fine, I may add some later :)

Also: it may be easier just to remove rather than comment-out, since they'll probably get stale if we don't maintain them?

**Review by charliermarsh — APPROVED** [review]
Thanks!

**Pull request merged**
